src/stactools/overture_maps/metadata.py

```diff
@@ -0,0 +1,446 @@
+            raise ValueError(f"could not parse partition info from {href}")
+
+        return cls(
+            href=href,
+            storage_backend=storage_backend,
+            release=parsed["release"],
+            theme=parsed["theme"],
+            feature_type=parsed["type"],
+            part=parsed["part"],
+            uid=parsed["uid"],
+            _datetime=datetime.strptime(parsed["release"][:10], "%Y-%M-%d"),
+        )
+
+    @property
+    def metadata(self) -> pq.FileMetaData:
+        """Parquet file metadata"""
+        if self._metadata is None:
+            pyarrow_args: PyarrowArgs
+
+            if self.storage_backend == StorageBackend.AZURE:
+                import adlfs
+
+                pyarrow_args = {
+                    "where": self.href.replace(f"https://{AZURE_NETLOC}/", ""),
+                    "filesystem": adlfs.AzureBlobFileSystem(
+                        **PYARROW_CONFIGS[self.storage_backend]
+                    ),
+                }
+            elif self.storage_backend == StorageBackend.AWS:
+                import s3fs
+
+                pyarrow_args = {
+                    "where": self.href,
+                    "filesystem": s3fs.S3FileSystem(
+                        **PYARROW_CONFIGS[self.storage_backend]
+                    ),
+                }
+            else:
+                pyarrow_args = {"where": self.href, "filesystem": None}
+
+            self._metadata = pq.read_metadata(**pyarrow_args)
+
+        return self._metadata
+
+    @property
+    def geo_metadata(self) -> Dict[str, Any]:
+        geo_metadata = json.loads(self.metadata.metadata[b"geo"].decode("utf-8"))
+        if not isinstance(geo_metadata, dict):
+            raise ValueError(f"could not parse geo metadata from {self.href}")
+
+        return geo_metadata
+
+    @property
+    def bbox(self) -> List[float]:
+        bbox: List[float] = self.geo_metadata["columns"]["geometry"]["bbox"]
+        if not all(isinstance(x, float) for x in bbox):
+            raise ValueError(
+                f"could not parse bounding box from {self.href}: {str(bbox)}"
+            )
+
+        return bbox
+
+    def to_item(self) -> Item:
+        collection = COLLECTION_ID_FORMAT.format(theme=self.theme)
+        asset = Asset(
+            href=self.href,
+            roles=["data"],
+            media_type=MediaType.PARQUET,
+        )
+
+        item = Item(
+            id=f"{self.theme}-{self.feature_type}-{self.release}-part-{self.part}-{self.uid}",
+            datetime=self._datetime,
+            bbox=self.bbox,
+            geometry=Polygon.from_bounds(*self.bbox).model_dump(exclude_none=True),
+            properties={
+                "overture:theme": self.theme,
+                "overture:type": self.feature_type,
+            },
+            assets={"data": asset},
+            collection=collection,
+        )
+
+        # version extension
+        VersionExtension.ext(item, add_if_missing=True)
```

Comment:
Is this supposed to be assigned to something? Or does this have side-effects?

Reply:
It has side effects, and the "more modern" way to do this is:

```suggestion
        item.ext.add("version");
```

The simpler extension interface was introduced in [v1.9.0](https://github.com/stac-utils/pystac/releases/tag/v1.9.0), led by @jsignell.